agent/agent.go

```diff
@@ -561,28 +580,6 @@ func (a *agent) init(ctx context.Context) {
 	}
 
 	go a.runLoop(ctx)
-	cl, err := a.client.AgentReportStats(ctx, a.logger, func() *codersdk.AgentStats {
-		stats := map[netlogtype.Connection]netlogtype.Counts{}
-		a.closeMutex.Lock()
```

Comment:
Review: The deadlock happens between https://github.com/coder/coder/blob/b10368517082fd77fabcf9c0791e92af5d3766c5/agent/agent_test.go#L898 and https://github.com/coder/coder/blob/b10368517082fd77fabcf9c0791e92af5d3766c5/agent/agent_test.go#L909.